js/components/Editor.js

```diff
@@ -59,7 +59,9 @@ const Editor = React.createClass({
   },
 
   componentDidUpdate() {
-    this.editor.setValue(this.props.sourceText);
```

Comment:
This and `componentWillReceiveProp` both need a lot of work. I'd say we mostly ignore this code for now until we focus on the editor. For some reason this code was giving me problems and it worked without it.